tools/generator/templates/player/caml/interface.cc.jinja2

```diff
@@ -0,0 +1,238 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (c) 2020 Association Prologin <association@prologin.org>
+{% import 'macros/cxx.jinja2' as cxx %}
+
+// This file contains the code to call the API functions from the OCaml
+// language.
+// {{ stechec2_generated }}
+
+#include <caml/mlvalues.h>
+#include <caml/callback.h>
+#include <caml/alloc.h>
+#include <caml/memory.h>
+
+#include <vector>
+#include <string>
+
+{% for enum in game.enum %}
+
+{{ cxx.decl_enum(enum) }}
+{% endfor %}
+{% for struct in game.struct %}
+
+{{ cxx.decl_struct(struct) }}
+{% endfor %}
+
+extern "C" {
+{% for func in game.function %}
+
+{{ func.fct_summary|cxx_comment(doc=True) }}
+{{ func|cxx_prototype(prefix='api_') }};
+{% endfor %}
+}
+
+template <typename CamlType, typename Cxx>
+CamlType cxx_to_caml(Cxx in)
+{
+    return in.__if_that_triggers_an_error_there_is_a_problem;
+}
+template <>
+value cxx_to_caml<value, int>(int in)
+{
+    {# TODO: investigate whether that empty call to CAMLparam0 really is
+     # necessary here and every time it's used in this file #}
+    CAMLparam0();
+    CAMLreturn(Val_int(in));
+}
+template <>
+value cxx_to_caml<value, double>(double in)
+{
+    CAMLparam0();
+    CAMLreturn(caml_copy_double(in));
+}
+template<>
+value cxx_to_caml<value, std::string>(std::string in)
+{
+    CAMLparam0();
+    size_t l = in.length();
+    char * out = (char *) malloc(l + 1);
+    for (size_t i = 0; i < l; i++) out[i] = in[i];
```

Comment:
Could you try to fix the formatting so that it matches our general C++ style? here you'd need a newline